File: app/src/main/java/ru/geekbrains/notes/view/Navigation.java
package ru.geekbrains.notes.view;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import ru.geekbrains.notes.R;

public class Navigation {

    private final FragmentManager fragmentManager;

    public Navigation(FragmentManager fragmentManager) {
        this.fragmentManager = fragmentManager;
    }

    public void addFragment(Fragment fragment, boolean useBackStack) {
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.notes_container, fragment);
        if(useBackStack) {
            fragmentTransaction.addToBackStack("");
        }
        fragmentTransaction.commit();
    }
}

File: app/src/main/java/ru/geekbrains/notes/view/Navigation.kt
package ru.geekbrains.notes.view

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import ru.geekbrains.notes.R

class Navigation(private val fragmentManager: FragmentManager) {
    fun addFragment(fragment: Fragment?, useBackStack: Boolean) {
        val fragmentTransaction = fragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.notes_container, fragment!!)
        if (useBackStack) {
            fragmentTransaction.addToBackStack("")
        }
        fragmentTransaction.commit()
    }
}